app/scripts/controllers/mmi-controller.ts

```diff
@@ -848,28 +849,30 @@ export default class MMIController extends EventEmitter {
   async setAccountAndNetwork(origin: string, address: string, chainId: number) {
     await this.appStateController.getUnlockPromise(true);
     const addressToLowerCase = address.toLowerCase();
-    const { address: selectedAddress } = this.messenger.call(
+    const { address: selectedAddress } = this.messagingSystem.call(
       'AccountsController:getSelectedAccount',
     );
 
     if (selectedAddress.toLowerCase() !== addressToLowerCase) {
-      const internalAccount = this.messenger.call(
+      const internalAccount = this.messagingSystem.call(
         'AccountsController:getAccountByAddress',
         addressToLowerCase,
       );
-      this.messenger.call(
-        'AccountsController:setSelectedAccount',
-        internalAccount.id,
-      );
+      if (internalAccount) {
+        this.messagingSystem.call(
+          'AccountsController:setSelectedAccount',
+          internalAccount.id,
+        );
+      }
     }
 
     const selectedChainId = getCurrentChainId({
-      metamask: this.networkController.state,
+      metamask: this.#networkControllerState,
     });
```

Comment:
It looks like previously, the entire NetworkController was being passed into the MMIController, and then the state of that controller was being obtained on demand. This means that the state was always live. Now we are  only getting the state of the NetworkController once, at initialization, and since the state gets replaced when it's updated, that means it could be stale here.

Since we already have the messenger can we get the currently selected chain ID from the NetworkController here? We can do that by saying:

``` suggestion
    const selectedNetworkClientId = await this.messagingSystem.call(
      'NetworkController:getState',
    ).selectedNetworkClientId;
    const selectedNetworkClient = await this.messagingSystem.call(
      'NetworkController:getNetworkClientById',
      selectedNetworkClientId,
    );
    const selectedChainId = selectedNetworkClient.configuration.chainId;
```

You may need to add `NetworkController:getNetworkClientById` to the list of allowed actions.